### PR TITLE
Fixed issue #16236: Vanilla survey theme options are not saved

### PR DESCRIPTION
**Pull request by gabrieljenik**
Vanilla is not using the core options page, and the custom one is not properly triggering the form submit.

### CODE REVIEW

File: themes/survey/vanilla/options/options.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ var ThemeOptions = function(){
             //now write the newly created object to the correspondent field as a json string
             $('#TemplateConfiguration_options').val(JSON.stringify(newOptionObject));
             //and submit the form
-            $('#template-options-form').find('button[type=submit]').trigger('click');
+            $('#template-options-form').trigger('submit');
```
Comment (Shnoulle) [inline themes/survey/vanilla/options/options.js:238]: With trigger('submit') : no HTML5 validation control is done (unsure it's needed here)

Comment (gabrieljenik) [reply themes/survey/vanilla/options/options.js:238]: I am not sure there is a button on that form.
I have run a quick search but couldn't find it.

Comment (Shnoulle) [reply themes/survey/vanilla/options/options.js:238]: Yes : i think here all is done via JS, then no need to validate form . Just a reminder :)

Comment (Shnoulle) [reply themes/survey/vanilla/options/options.js:238]: No button , alt solution : add a button in form ?

Comment (gabrieljenik) [reply themes/survey/vanilla/options/options.js:243]: Comment added in the code to explain "no-button" situation. Will be helpfull for future.

Comment (sushmanadendlalimesurvey) [reply themes/survey/vanilla/options/options.js:243]: Tested issue by pulling the PR. Able to save the changes successfully

**Comment by Shnoulle** [comment]
PS : thank you : https://gitlab.com/SondagesPro/SurveyThemes/skelvanilla/-/commit/209da6bd270d0a118efffba7afcd37079f45a97f

**Comment by sushmanadendlalimesurvey** [comment]
Tested issue by pulling the PR. Able to save the changes successfully

**Pull request merged**